### PR TITLE
MG-687: Add age_numeric to disease_correlation

### PR DESCRIPTION
**Pull request by beatrizsaldana**
# Add `age_numeric` Field to Disease Correlation Transform

## Problem

The disease correlation dataset contains an `age` field stored as a string (e.g., "4 months", "8 months", "12 months"). This format makes it difficult to:
- Perform numeric operations or sorting on age values
- Filter or query data by age ranges efficiently
- Use age as a numeric parameter in downstream analyses

Additionally, other transforms in the codebase may need similar functionality to extract numeric values from formatted age strings.

### 1. Added `age_numeric` Field
- Added a new `age_numeric` field to the disease correlation transform output
- The field contains the integer value extracted from the `age` string (e.g., "8 months" → 8)
- Positioned directly after the `age` field in the output structure for logical grouping

### 2. Created Reusable Utility Function
- Implemented `extract_age_numeric()` function in `src/agoradatatools/etl/utils.py`
- Uses regex to extract the first numeric value from an age string
- Returns 0 if no numeric value is found
- Well-documented with examples for easy reuse across other transforms

### 3. Updated Transform Logic
- Modified `process_group()` in `disease_correlation.py` to include the new field
- Updated function documentation to reflect the new output structure
- Imported `extract_age_numeric()` from utils instead of defining locally

### Unit Tests
- **7 new test cases** for `extract_age_numeric()` covering:
  - Standard age formats: "4 months", "8 months", "12 months"
  - Alternative time units: "6 weeks", "100 days"
  - Edge cases: empty strings, strings without numbers
- **Updated all existing tests** in disease_correlation to include `age_numeric` field
- **Test asset integration tests** updated with new field expectations

## Example Output
```
{
  "name": "5xFAD",
  "cluster": "Consensus Cluster A (ECM organization)",
  "age": "4 months",
  "age_numeric": 4,
  "sex": "Female",
  "IFG": {
    "correlation": 0.0829228557717283,
    "adj_p_val": 0.4617459764146099
  }
}
```

### CODE REVIEW

File: tests/test_utils.py [hunk 1]
```diff
@@ -887,3 +887,33 @@ def test_negative_zero_via_copysign(self) -> None:
...
+    @pytest.mark.parametrize(
+        "input_age,expected",
+        [
+            ("4 months", 4),  # Age with months
+            ("8 months", 8),  # Age with months
+            ("12 months", 12),  # Age with months
+            ("6 weeks", 6),  # Age with weeks
+            ("100 days", 100),  # Age with days
+            ("", None),  # Empty string
+            ("no number here", None),  # String without number
```
Comment (jaclynbeck-sage) [inline tests/test_utils.py:907]: Two more tests: 
("number 10", None) <-- I believe returning None should be the correct behavior here, when the number is not the first thing in the string. 
("10 11 12", ??) <-- This should maybe just return 10? And assume the function returns the first number in the string by design? I think the real "desired" behavior would be to throw an error if multiple numbers are found but I don't know how much extra work/extra code it would be to add that check and don't think it's worth putting too much time into it.

Comment (jaclynbeck-sage) [reply tests/test_utils.py:907]: Oh, a third test, (None, None).

**Review by jaclynbeck-sage — APPROVED** [review]
Approving! Just add the 3 extra test cases, and if it goes horribly wrong and you have to add more code I'll re-review. But I *think* with the way you've written it, it should be ok.

File: tests/test_utils.py (hunk 1 above)
Comment (jaclynbeck-sage) [reply tests/test_utils.py:907]: Oh ok, fiddling around with re, I think you need to either a) change the expression to `r"$(\d+)"` if we only return an age for strings that start with a number, or b) just assume (and document) that the function will return the first number it finds anywhere in the string. Either solution is fine with me. 
And it looks like you'll need to add a check for age == None in the function.
Otherwise I think we're good!

Comment (JessterB) [reply tests/test_utils.py:907]: > Two more tests: ("number 10", None) <-- I believe returning None should be the correct behavior here, when the number is not the first thing in the string. ("10 11 12", ??) <-- This should maybe just return 10? And assume the function returns the first number in the string by design? I think the real "desired" behavior would be to throw an error if multiple numbers are found but I don't know how much extra work/extra code it would be to add that check and don't think it's worth putting too much time into it.

I think it's fine to assume that the function returns the first number in the string by design. We should end up catching malformatted output values like "age": "12 10 4 months" in GX anyway.

Comment (jaclynbeck-sage) [reply tests/test_utils.py:907]: Cool, then the three test cases should be: 
("number 10", 10)
("10 11 12", 10)
(None, None)
Thanks Jess!

Comment (beatrizsaldana) [reply tests/test_utils.py:907]: Done!
